src/components/userProfile/sagas.js

```diff
@@ -0,0 +1,54 @@
+import { effects } from 'redux-saga';
+import {  REQUEST_AUTH_USER_JWT } from './actionTypes';
+import { SIGN_URL } from './constants';
+import app from '../app';
+
+const { actions: { userDataNotFound, getUserData } } = app;
+
+/**
+ * Makes a request to sign up a user
+ *
+ * @param {object} [action] The data passed from the watcher generator
+ */
+function* getAllUserInfo(action) {
+    try {
+        const { payload } = action;
```

Comment:
`const { idToken } = action.payload;`

Reply:
here, `payload` is not an object. it is the actual value needed

Reply:
that is what I am trying to say it should be an object.

Reply:
Ok, cool